### PR TITLE
Fixed issue #153, escape queries generated by Riak store.

### PR DESCRIPTION
**Pull request by cabol**
- Fixed issue #153, escape queries generated by Riak store.
- Function `build_query` was totally modified to support different scenarios, not only escape string values but also handle conditions such as: `<, <=, >, >=, /=, ==, null, not_null`.
- Fixed bug in find_by function, when argument 'Conditions' is a tuple (specifying only one condition), instead of a list.

### CODE REVIEW

**Comment by elbrujohalcon** [comment]
@cabol to make sure your changes are actually working, you should remove riak from [here](https://github.com/inaka/sumo_db/blob/master/test/sumo_test_utils.erl#L28) and run the tests

File: test/conditional_logic_riak_SUITE.erl [hunk 1]
```diff
@@ -0,0 +1,203 @@
...
+
+  Module = sumo_test_people_riak,
+
+  sumo:create_schema(Module),
+  sumo:delete_all(Module),
+
+  sumo:persist(Module, Module:new("Jane", "Doe")),
+  sumo:persist(Module, Module:new("John", "Doe", 30, "2015-01-01")),
+  sumo:persist(Module, Module:new("Jane Jr.", "Doe", 5, "2015-01-05")),
+  sumo:persist(Module, Module:new("Joe", "Armstrong")),
```
Comment (gadgetci) [inline test/conditional_logic_riak_SUITE.erl]: According to **Elvis**:
> Remove the dynamic function call on line 49. Only modules that define callbacks should make dynamic calls.

File: test/conditional_logic_riak_SUITE.erl [hunk 2]
```diff
@@ -0,0 +1,203 @@
...
+  sumo_test_utils:start_apps(),
+
+  Module = sumo_test_people_riak,
+
+  sumo:create_schema(Module),
+  sumo:delete_all(Module),
+
+  sumo:persist(Module, Module:new("Jane", "Doe")),
+  sumo:persist(Module, Module:new("John", "Doe", 30, "2015-01-01")),
+  sumo:persist(Module, Module:new("Jane Jr.", "Doe", 5, "2015-01-05")),
```
Comment (gadgetci) [inline test/conditional_logic_riak_SUITE.erl]: According to **Elvis**:
> Remove the dynamic function call on line 48. Only modules that define callbacks should make dynamic calls.

File: test/conditional_logic_riak_SUITE.erl [hunk 3]
```diff
@@ -0,0 +1,203 @@
...
+-spec init_per_suite(config()) -> config().
+init_per_suite(Config) ->
+  sumo_test_utils:start_apps(),
+
+  Module = sumo_test_people_riak,
+
+  sumo:create_schema(Module),
+  sumo:delete_all(Module),
+
+  sumo:persist(Module, Module:new("Jane", "Doe")),
```
Comment (gadgetci) [inline test/conditional_logic_riak_SUITE.erl]: According to **Elvis**:
> Remove the dynamic function call on line 46. Only modules that define callbacks should make dynamic calls.

File: test/conditional_logic_riak_SUITE.erl [hunk 4]
```diff
@@ -0,0 +1,203 @@
...
+  sumo:persist(Module, Module:new("John", "Doe", 30, "2015-01-01")),
+  sumo:persist(Module, Module:new("Jane Jr.", "Doe", 5, "2015-01-05")),
+  sumo:persist(Module, Module:new("Joe", "Armstrong")),
+  sumo:persist(Module, Module:new("Alan", "Turing", 102, "2015-01-10")),
```
Comment (gadgetci) [inline test/conditional_logic_riak_SUITE.erl]: According to **Elvis**:
> Remove the dynamic function call on line 50. Only modules that define callbacks should make dynamic calls.

File: test/conditional_logic_riak_SUITE.erl [hunk 5]
```diff
@@ -0,0 +1,203 @@
...
+init_per_suite(Config) ->
+  sumo_test_utils:start_apps(),
+
+  Module = sumo_test_people_riak,
+
+  sumo:create_schema(Module),
+  sumo:delete_all(Module),
+
+  sumo:persist(Module, Module:new("Jane", "Doe")),
+  sumo:persist(Module, Module:new("John", "Doe", 30, "2015-01-01")),
```
Comment (gadgetci) [inline test/conditional_logic_riak_SUITE.erl]: According to **Elvis**:
> Remove the dynamic function call on line 47. Only modules that define callbacks should make dynamic calls.

**Comment by cabol** [comment]
@elbrujohalcon now is ready to merge!!

File: test/conditional_logic_SUITE.erl [hunk 6]
```diff
@@ -47,7 +47,12 @@ init_per_suite(Config) ->
         sumo:persist(Module, Module:new("John", "Doe", 30)),
         sumo:persist(Module, Module:new("Jane Jr.", "Doe", 5)),
         sumo:persist(Module, Module:new("Joe", "Armstrong")),
-        sumo:persist(Module, Module:new("Alan", "Turing", 102, "Computer St."))
+        sumo:persist(Module, Module:new("Alan", "Turing", 102, "Computer St.")),
+
+        case Module of
+          sumo_test_people_riak -> timer:sleep(5000);
```
Comment (elbrujohalcon) [inline test/conditional_logic_SUITE.erl]: I'm very aware that you will hate me… even more… after this comment but: can you keep the style and create a `sumo_test_utils:sleep_if_required(Module)` and put this case (in a function-head pattern-matching fashion) in it?
That way we will not need to have cases and conditionals checking module names in the tests themselves.

**Pull request merged**